apps/wallet/src/ui/app/components/ledger/ImportLedgerAccounts.tsx

```diff
@@ -1,41 +1,112 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { LockUnlocked16 as UnlockedLockIcon } from '@mysten/icons';
-import { Navigate, useNavigate } from 'react-router-dom';
+import {
+    LockUnlocked16 as UnlockedLockIcon,
+    Spinner16 as SpinnerIcon,
+    ThumbUpStroke32 as ThumbUpIcon,
+} from '@mysten/icons';
+import { useCallback } from 'react';
+import toast from 'react-hot-toast';
+import { useNavigate } from 'react-router-dom';
 
 import { SummaryCard } from '../SummaryCard';
 import { useNextMenuUrl } from '../menu/hooks';
 import Overlay from '../overlay';
-import { LedgerAccount } from './LedgerAccount';
-import { useSuiLedgerClient } from '_src/ui/app/components/ledger/SuiLedgerClientProvider';
+import { type LedgerAccount } from './LedgerAccountItem';
+import { SelectLedgerAccountsList } from './SelectLedgerAccountsList';
+import { useDeriveLedgerAccounts } from './useDeriveLedgerAccounts';
 import { Button } from '_src/ui/app/shared/ButtonUI';
 import { Link } from '_src/ui/app/shared/Link';
+import { Text } from '_src/ui/app/shared/text';
 
-const mockAccounts = [
-    {
-        isSelected: false,
-        address:
-            '0x7a286c8455a801f6d81faaa0f87543fa4a0de64dcc48b9c9308ee18f0f6ccdd3',
-        balance: 30,
-    },
-    {
-        isSelected: true,
-        address:
-            '0x7a286c8455a401f6d81faaa0f87543fa4a0de64dcc48b9c9308ee18f0f6ccdd3',
-        balance: 30000,
-    },
-];
+const numLedgerAccountsToDeriveByDefault = 10;
```

Comment:
is this just a sensible default? just curious where 10 comes from

Reply:
see my comment above!